### PR TITLE
Introducing some nullable ref type support on public APIs

### PR DESCRIPTION
**Pull request by danielmarbach**
Not all public APIs are done but this is a good first step.

We could also aim for covering all the public types in the Approval file but I figured we have to get started "somewhere"

### CODE REVIEW

File: src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace NServiceBus
     public class AssemblyScannerConfiguration
     {
         public AssemblyScannerConfiguration() { }
-        public string AdditionalAssemblyScanningPath { get; set; }
+        public string? AdditionalAssemblyScanningPath { get; set; }
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt:17]: This is not a breaking change, right?

Comment (danielmarbach) [reply src/NServiceBus.Core.Tests/ApprovalFiles/APIApprovals.ApproveNServiceBus.approved.txt:17]: No. Without nullable reference type this property had to be assumed to potentially contain null. Even our internal code checks for that.

**Comment by danielmarbach** [comment]
@DavidBoike the usage of the guard class requires an update to make this PR compile

**Comment by DavidBoike** [comment]
@danielmarbach Yep, adjusted that, and added an approval test to list types that do not have nullable annotations. That list should shrink over time, growth should not be approved.

**Comment by danielmarbach** [comment]
Nice!

**Pull request merged**